### ISSUE TITLE
Gitlab Pipeline Results are reported unreliably

### ISSUE THREAD

**Issue by FlorianBorn** [MicroFocus/octane-gitlab-service#103]
Hello,

we noticed that sometimes pipeline results are not reported to Octane (in Octane the pipeline is stuck in state 'running'). 
After starting a pipeline we can see the pipeline to appear running in Octane, but after the pipeline successfully finished in gitlab it  never appears as finished in Octane. Test-Results are also not reported.

What could cause this issue?

<details>
<summary>Logs</summary>

```  
[INFO ] 2023-11-24 12:21:55.447 [AbridgedTasksWorker-279] BridgeServiceImpl - [https://dummy.de?p=5001] result for task 'a4ac706c-01cd-4736-b8a5-ace07f39a953' submitted with status 204
[WARN ] 2023-11-24 12:29:57.640 [qtp1811922029-18] EventListener - Failed to return the 'started_at' of the job, using null as default.
[WARN ] 2023-11-24 12:29:57.640 [qtp1811922029-18] EventListener - Failed to return the 'finished_at' of the job, using null as default.
[WARN ] 2023-11-24 12:30:01.279 [qtp1811922029-18] EventListener - Failed to infer event cause type, using 'USER' as default
[INFO ] 2023-11-24 12:30:01.794 [qtp1811922029-18] ConfigurationServiceImpl - [https://dummy.de?p=5001] addToOctaneRootsCache: new root is added [pipeline:dummy_gitlab/repo/dummy_branch]
[INFO ] 2023-11-24 12:30:02.474 [EventsServiceWorker-66] EventsServiceImpl - [https://dummy.de?p=5001] sending [pipeline:dummy_gitlab/repo/dummy_branch:25953831:STARTED] event/s. Correlation ID - 78316113240c44d48dac8fa54
[WARN ] 2023-11-24 12:30:03.233 [qtp1811922029-18] EventListener - Failed to return the SCM data. Returning null.
[INFO ] 2023-11-24 12:30:55.262 [qtp1811922029-18] EventListener - Merge Request event occurred.
[WARN ] 2023-11-24 12:32:19.943 [qtp1811922029-20] EventListener - Failed to return the 'started_at' of the job, using null as default.
[WARN ] 2023-11-24 12:32:19.944 [qtp1811922029-20] EventListener - Failed to infer event cause type, using 'USER' as default
[INFO ] 2023-11-24 12:32:20.503 [EventsServiceWorker-66] EventsServiceImpl - [https://dummy.de?p=5001] sending [dummy_gitlab/repo/POC_Appium:170393532:STARTED, pipeline:dummy_gitlab/repo/dummy_branch:25953831:FINISHED] event/s. Correlation ID - f3b7593735ed4909b21c1a3d0
```  
</details>

**Comment by TiutiuMadalin** [comment]
Hi @FlorianBorn, we found this issue as well we are currently testing a fix for it

**Comment by TiutiuMadalin** [comment]
@FlorianBorn Just to check that, the issue is the same, can you confirm that only the last job remains stuck in running (in this case POC_Appium)?

**Comment by FlorianBorn** [comment]
Hi @TiutiuMadalin,
thanks for your reply. I just checked some pipeline jobs and can confirm that.

**Comment by TiutiuMadalin** [comment]
fixed with [v23.4.0-CE](https://github.com/MicroFocus/octane-gitlab-service/releases/tag/v23.4.0-CE)
